GEMINI.md

```diff
@@ -0,0 +1,151 @@
+      * If they are in the `parser` module, you _should_ add traditional unit tests with a test proto if possible, you _may_ add golden unit tests to easily see the changes. For example, [routing_rule_parser_testing](https://github.com/googleapis/sdk-platform-java/blob/main/gapic-generator-java/src/test/proto/routing_rule_parser_testing.proto) that is only used for testing [RoutingRuleParser](https://github.com/googleapis/sdk-platform-java/blob/main/gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/RoutingRuleParser.java).
+      * If they are in `other modules(ast, model, writer etc.)`, you _must_ add traditional unit tests, you _may_ add golden unit tests to easily see the changes.
+- If the changes are in both `gax` and `gapic-generator-java`, you _must_ add all test layers, including traditional unit tests, golden unit tests and showcase integration tests.
+
+### 4.2. Running Unit Tests
+
+Unit tests can be tested via this command:
+
+```sh
+mvn test
+```
+
+### 4.3. Running Golden Integration Tests
+
+Golden integration tests are run using Bazel. To run all golden integration tests, use the following command from the root of the repository:
+
+```sh
+bazelisk test //...
+```
+
+Specific integration tests can be run by specifying the target, for example:
+
+```sh
+bazelisk test //test/integration:redis
+```
+
+### 4.4. Updating Golden Files
+
+If you make changes that affect the generated code, you will need to update the golden files. This can be done using the following command:
+
+```sh
+bazelisk run //test/integration:update_asset && bazelisk run //test/integration:update_credentials && bazelisk run //test/integration:update_iam && bazelisk run //test/integration:update_kms && bazelisk run //test/integration:update_pubsub && bazelisk run //test/integration:update_logging && bazelisk run //test/integration:update_redis && bazelisk run //test/integration:update_storage && bazelisk run //test/integration:update_library && bazelisk run //test/integration:update_compute && bazelisk run //test/integration:update_bigtable && bazelisk run //test/integration:update_apigeeconnect 
+```
+
+### 4.5. Running Showcase Integration Tests
+
+Showcase integration tests are run against a local server that implements the Showcase API. The `java-showcase/README.md` file provides detailed instructions on how to run these tests. The general steps are:
+
+1.  **Install the Showcase server:**
+
+    ```sh
+    go install github.com/googleapis/gapic-showcase/cmd/gapic-showcase@latest
+    ```
+
+2.  **Run the Showcase server:**
+
+    ```sh
+    gapic-showcase run
+    ```
+
+3.  **Run the integration tests:**
+
+    ```sh
+    cd java-showcase
+    mvn verify -P enable-integration-tests
+    ```
+    
+## 5. Dependency Management
+
+- Try not to bump any dependency version unless there is a known CVE (security or vulnerability issue)
```

Comment:
I think there are three main reasons to bump dependency versions

- CVE
- Critical bug fixes
- Required new features

Reply:
Adding the first two to the file. I think `required new features` may be a bit too vague. I can play around with the wording.